include/irr/asset/IAssetManager.h

```diff
@@ -422,8 +422,7 @@ class IAssetManager : public core::IReferenceCounted, public core::QuitSignallin
         //TODO change name
         bool insertAssetIntoCache(SAssetBundle& _asset)
         {
-            const uint32_t ix = IAsset::typeFlagToIndex(_asset.getAssetType());
-            return m_assetCache[ix]->insert(_asset.getCacheKey(), _asset);
+            return insertAssetIntoCache(_asset, IAsset::EM_CPU_PERSISTENT);
```

Comment:
this is now mutable by default, yes?

Reply:
public `IAssetManager::insertIntoCache` sets asset's mutability as cpu persistent (so that everything that is added to cache explicitely by user is cpu persistent), but `IAssetLoaderOverride` sets mutability as mutable (everything loaded by `getAsset` and subassets that are added to cache during the loading process are mutable)
All the builtins are added to cache as immutable

Reply:
ok I guess it kind-of makes sense.

Reply:
not entirely convinced about the asset manager default behaviour.

Reply:
hmm ok can keep as is.

Reply:
but I'd instead add a `mutability` argument to the function with ` IAsset::EM_CPU_PERSISTENT` as the default parameter so user doesn't have to do the long form of first inserting an asset and then changing its mutability to what he actually wants